pkg/karmadactl/promote/promote.go

```diff
@@ -149,6 +164,7 @@ func (o *CommandPromoteOption) AddFlags(flags *pflag.FlagSet) {
 		"Context name of legacy cluster in kubeconfig. Only works when there are multiple contexts in the kubeconfig.")
 	flags.StringVar(&o.ClusterKubeConfig, "cluster-kubeconfig", "",
 		"Path of the legacy cluster's kubeconfig.")
+	flags.BoolVarP(&o.Deps, "deps", "d", true, "Promote resource with its dependencies automatically, default to true")
```

Comment:
I think it's better to use the full name: `flags.BoolVarP(&o.Deps, "dependencies", "d", ***`